docs/ProviderDesign.md

```diff
@@ -0,0 +1,79 @@
+# Provider's view of the FLOCX
+
+![ProviderDesign](../images/ProviderService-Marketplace.png)
+
+
+
+
+## Multi-tenant IRONIC:
+Ironic manages multiple pools of hardware each owned by a separate user/group/organization.
+
+
+## Provider Service:
+A centralized service per OpenStack deployment that manages offers and contracts 
+on behalf of all its tenants of the OpenStack.
+
+## Keystone:
+Provides authentication and authorization services.
+
+
+## Workflow:
+
+Following steps describe the process of offering physical hardware in the market by individual owners
+and receiving contracts from the marketplace.
+
+
+
+1. Hardware pool owners decide which nodes to offer in the marketplace and submit offers to the provider service
+2. Provider service confirms owner-hardware relationship before accepting offers. 
+3. For each accepted offer it fetches the hardware configuration from Ironic
+4. These offers are published to the marketplace as soon as possible. 
+5. Receives contracts from the marketplace.
+
+
+## Offers (Owners --> Providers):
+An offer made by an owner and submitted to the provider service will have following attributes.
+* Node ID: Unique ID of the node
+* Start time: Date and time of availability
+* Duration: Hours, days that the node will be available
+* Floor Price: Minimum Price that is demanded.
+
+Provider validates the offer received from the owner  from keystone and adds the following attributes:
```

Comment:
I get what you're saying, but I don't think this is quite precise. The owner will not be able to submit an offer through the provider API unless they're already authenticated in keystone. What the provider will do is verify that the owner matches the owner listed for the node.

Reply:
made the changes.